### PR TITLE
[#DEV-7321] Adds TimerGPU

### PR DESCRIPTION
**Pull request by Kuranes**
Adds a Timer GPU using webgl extenstion https://www.khronos.org/registry/webgl/extensions/EXT_disjoint_timer_query/

Have to call it before and after a set of gl calls to get a timer.
```js
TimerGPU.instance().start('mypostproc');
glcalls...
TimerGPU.instance().end('mypostproc');
```

### CODE REVIEW

File: sources/osg/TimerGPU.js [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+        if ( !this._hasTimeElapsed && !this.hasTimeStamp ) {
+            return;
+        }
+
+        // no timestamp means not start/end absolute time
+        // which means each start must be followed by a end
+        // BEFORE any other start (of other queryID)
+        if ( !this.hasTimeStamp ) {
+            console.log( 'Warning: do not use interleaved GPU query' );
```
Comment (cedricpinson) [inline sources/osg/TimerGPU.js]: osg.warn

File: sources/osg/TimerGPU.js [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+    // query waiting async results from GPU 
+    this._waitingQueries = {};
+};
+
+TimerGPU.instance = function ( gl ) {
+
+    if ( !TimerGPU._instance ) {
+        TimerGPU._instance = new TimerGPU( gl );
+    } else if ( gl && TimerGPU._instance.getContext() !== gl ) {
+        TimerGPU._instance.setContext( gl );
```
Comment (cedricpinson) [inline sources/osg/TimerGPU.js]: why ?

**Comment by cedricpinson** [comment]
seems good, Could you add this into a osgjs sample ? like shadow maybe ?

File: sources/osg/TimerGPU.js (hunk 1 above)
Comment (Kuranes) [inline sources/osg/TimerGPU.js]: k

File: sources/osg/TimerGPU.js (hunk 2 above)
Comment (Kuranes) [inline sources/osg/TimerGPU.js]: In case first call to timerGPU.instance() is without the gl parameter. 
Cannot remember how that did happen.
Will remove.

**Comment by Kuranes** [comment]
Miss a way to use those in GL code.
Either do some renderstage override ?
OR my preferred solution add a osg:Camera:beforeFirstDrawCall() callback mechanism where we could put the timerGPU::start
(we already have the osg:Camera:FinalDrawCall() https://github.com/cedricpinson/osgjs/blob/develop/sources/osg/Camera.js#L74 )

**Comment by cedricpinson** [comment]
Check in osg the different callback and how they are called https://github.com/cedricpinson/osg/blob/master/include/osg/Camera#L611

**Comment by cedricpinson** [comment]
Seems like to leak, it would be more appropriate to have a number of frame and using a modulo to reuse same entry in map/array

**Comment by Kuranes** [comment]
- fixed leaks
- rebased

**Pull request merged**